Chime/Parser.m

```diff
@@ -0,0 +1,268 @@
+#import "Headers/Parser.h"
+#import "Headers/Opcode.h"
+#import "Headers/Utilities.h"
+
+#define IS_LABEL YES
+#define IS_NOT_LABEL NO
+
+@implementation Parser
+
+- (Parser *)init {
+  self = [super init];
+  if (self) {
+    _program = [[NSArray alloc] init];
+    _labels = [[NSMapTable alloc] init];
+    _variables = [[NSMapTable alloc] init];
+    _keywords = [[NSMapTable alloc] init];
+  }
```

Comment:
Shouldn't we initialize the counter here? I know we explicitly set it to 0 at the beginning of each pass, but this looks like an accident waiting to happen.